src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -183,6 +194,96 @@ object ExtractFilterNode {
 
     case _ => None // plan does not match with any of filter index rule patterns
   }
+
+  def extractNamesFromExpression(exp: Expression): Set[String] = {
```

Comment:
Seems these util functions will also be used for join rule. These also can be moved to the new util file.